### PR TITLE
Allow multi-language code samples

### PR DESCRIPTION
**Pull request by eriwen**
Functionality:
multi-lang-sample

- User selection is remembered in `window.localStorage` by language
- Does nothing if only one language sample provided
- Works in IE9+
- Default title is language if not provided

How to use:

- Samples in multiple language must be right next to each other
- Must provide a language or sample title
- Only 1 set of languages is supported

```
[source.multi-language-sample,groovy]
.build.gradle
----
include::{samplescodedir}/copy.gradle[]
----

[source.multi-language-sample,kotlin]
.build.gradle.kts
----
include::{samplescodedir}/copy.gradle.kts[]
----
```

### CODE REVIEW

File: gradle/local-publish.gradle [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apply plugin: 'maven-publish'
+
+publishing {
+    publications {
+        maven(MavenPublication) {
+            from components.java
+        }
+    }
+}
```
Comment (eskatos) [inline gradle/local-publish.gradle]: ❓ why not use composite builds for trying out the plugin locally?

File: build.gradle [hunk 2]
```diff
@@ -8,14 +8,19 @@ plugins {
...
 
 apply from: "$rootDir/gradle/functional-test.gradle"
+apply from: "$rootDir/gradle/local-publish.gradle"
 
 buildScan {
     licenseAgreementUrl = 'https://gradle.com/terms-of-service'
     licenseAgree = 'yes'
+    if (!System.getenv("CI")) {
+        publishAlways()
+        tag("CI")
```
Comment (eskatos) [inline build.gradle]: ❓ `if(!CI) tag("CI")` doesn't make much sense to me, mistake?

Comment (bamboo) [reply build.gradle]: By the way, time to migrate to the Kotlin DSL?

**Comment by JLLeitschuh** [comment]
Just a note about your example:
You might want to use the assignment syntax instead.
```kotlin
tasks {
    "copy"(Copy::class) {
        group = "Custom"
        description = "Copies the source to the destination directory"
    }
}
```

File: gradle/local-publish.gradle (hunk 1 above)
Comment (eriwen) [reply gradle/local-publish.gradle]: I have been, but I need build scans for debugging and until recently this did not work. We can omit this file if you prefer

File: build.gradle (hunk 2 above)
Comment (eriwen) [reply build.gradle]: Yes, a mistake. Should remove `!`. I've been migrating guides slowly to Kotlin DSL. Note that how we have the guides project setup right now, all our Kotlin build scripts go red. I may need your guidance help with this later.

**Comment by eriwen** [comment]
@JLLeitschuh @eskatos @bamboo We should prefer one syntax across all our docs, and plan for IDEs to warn users using others. Please tell me which syntax we should use for common constructs like task declaration, task configuration, plugin application, and so on. This is critical, please do this as soon as possible so that when we start providing samples, they are consistent.

**Comment by eriwen** [comment]
I have filed https://github.com/gradle/kotlin-dsl/issues/818 and marked it as blocking converting samples.

**Comment by eriwen** [comment]
Another look @eskatos? Once this is merged and we know what Kotlin syntax to prefer, folks can start converting guides examples.

**Pull request merged**